code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -368,12 +370,20 @@
 				success = TRUE
 
 		if(success)
-			F = F.make_plating(TRUE) || F
-			if(was_replacing && tilestack)
+			var/area/is_this_maints = get_area(F)
+			if(was_replacing && tilestack)	//turn the tile into plating (if needed), then replace it
+				F = F.make_plating(TRUE) || F
 				tilestack.place_tile(F)
 				if(!tilestack)
 					speak("Requesting refill of custom floor tiles to continue replacing.")
-			else
+			else if(F.broken || F.burnt)	//repair the tile and reset it to be undamaged (rather than replacing it)
+				F.broken = FALSE
+				F.burnt = FALSE
+				F.icon_state = initial(F.icon_state)
+			else if(is_this_maints.area_flags == IS_MAINTENANCE_AREA)	//place catwalk if it's plating and we're in maints
```

Comment:
`area_flags` is a bitflield, which means it's effectively one big number, and we're individually flipping its bits to store true/false stuff, so there's a bit that's for maintenance, if it's flipped on we're maint, if it's not we're not, etc.

You can't check that with just ==, since it will test if the whole flags number is equal to 1 set bit. That'll fail, since we store more then just that in `area_flags`.

You'll want to use binary operators. & will check if two numbers share any bitflags, so if you & `area_flags` and `IS_MAINTENANCE_AREA` together you'll be able to see if `area_flags` has `IS_MAINTENANCE_AREA` set.

You'll want to read this wiki page https://tgstation13.org/wiki/Binary_flags
It gives a really pretty decent rundown on the idea. Only thing to note is 1<<1 is setting the second bit to 1, 1<<5 sets the fifth. That's not discussed in the wiki page I think because it didn't used to work? They use the raw numbers instead, so 1, 2, 4, 8, 16, 32, etc.

If anythings confusing feel free to bug me on discord or whatever

Reply:
Yeah, I think I get it. Was wondering what the 1 < # meant but that explains it pretty well.
I _think_ I wrote it correctly, call me out if I did it wrong but I at least understand what it's meant to be doing now

Reply:
Ya did good